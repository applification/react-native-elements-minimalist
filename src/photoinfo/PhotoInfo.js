import React, { Component } from 'react';
import { View, Text, Image } from 'react-native';
import PropTypes from 'prop-types';

import Profile from '../profile/Profile';

class PhotoInfoComponent extends Component {
  static propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string,
    photo: PropTypes.any,
    icon: PropTypes.any,
  };

  render() {
    const { icon, title, photo, subtitle } = this.props;
    return (
      <View
        style={{
          flexDirection: 'row',
          justifyContent: 'flex-start',
          paddingLeft: 3,
          paddingTop: 5,
          paddingBottom: 5,
          height: 50,
        }}
      >
        {photo ? (
          <Profile small source={photo} style={{ alignSelf: 'flex-start' }} />
        ) : null}

        {title && subtitle ? (
          <Text
            style={{
              alignSelf: 'center',
              alignContent: 'flex-start',
              flexGrow: 1,
              paddingLeft: 7,
            }}
          >
            {title}
            {'\n'}
            {subtitle}
          </Text>
        ) : (
          <Text
            style={{
              alignSelf: 'center',
              alignContent: 'flex-start',
              flexGrow: 1,
              paddingLeft: 7,
            }}
          >
            {title}
          </Text>
        )}

        <Image
          source={icon}
          style={{
            alignSelf: 'center',
            marginLeft: 10,
            marginRight: 10,
            height: 22,
            width: 22,
          }}
        />
      </View>
    );
  }
}

export default PhotoInfoComponent;
