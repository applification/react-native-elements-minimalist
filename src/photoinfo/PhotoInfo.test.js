import React from 'react';
import renderer from 'react-test-renderer';

import PhotoInfo from './PhotoInfo';

describe('PHOTOINFO', () => {
  it('Renders photo info component', () => {
    const component = renderer.create(
      <PhotoInfo
        photo={{
          uri:
            'https://s3.amazonaws.com/uifaces/faces/twitter/kfriedson/128.jpg',
        }}
        title="josh.duplessis"
        icon={require('../imgs/more.png')}
      />
    );

    expect(component).toMatchSnapshot();
  });

  it('Renders WITHOUT out subtitle', () => {
    const component = renderer.create(
      <PhotoInfo
        photo={{
          uri:
            'https://s3.amazonaws.com/uifaces/faces/twitter/kfriedson/128.jpg',
        }}
        title="josh.duplessis"
        icon={require('../imgs/more.png')}
      />
    );

    expect(component).toMatchSnapshot();
  });

  it('Renders WITH a subtitle', () => {
    const component = renderer.create(
      <PhotoInfo
        photo={{
          uri:
            'https://s3.amazonaws.com/uifaces/faces/twitter/kfriedson/128.jpg',
        }}
        title="bentierney"
        subtitle="Guitar Center"
        icon={require('../imgs/more.png')}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
