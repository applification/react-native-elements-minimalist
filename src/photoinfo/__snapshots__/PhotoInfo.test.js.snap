// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PHOTOINFO Renders WITH a subtitle 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "flex-start",
      "paddingBottom": 5,
      "paddingLeft": 3,
      "paddingTop": 5,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://s3.amazonaws.com/uifaces/faces/twitter/kfriedson/128.jpg",
      }
    }
    style={
      Object {
        "borderRadius": 15,
        "height": 30,
        "margin": 7,
        "width": 30,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignContent": "flex-start",
        "alignSelf": "center",
        "flexGrow": 1,
        "paddingLeft": 7,
      }
    }
  >
    bentierney
    

    Guitar Center
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../src/imgs/more.png",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "height": 22,
        "marginLeft": 10,
        "marginRight": 10,
        "width": 22,
      }
    }
  />
</View>
`;

exports[`PHOTOINFO Renders WITHOUT out subtitle 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "flex-start",
      "paddingBottom": 5,
      "paddingLeft": 3,
      "paddingTop": 5,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://s3.amazonaws.com/uifaces/faces/twitter/kfriedson/128.jpg",
      }
    }
    style={
      Object {
        "borderRadius": 15,
        "height": 30,
        "margin": 7,
        "width": 30,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignContent": "flex-start",
        "alignSelf": "center",
        "flexGrow": 1,
        "paddingLeft": 7,
      }
    }
  >
    josh.duplessis
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../src/imgs/more.png",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "height": 22,
        "marginLeft": 10,
        "marginRight": 10,
        "width": 22,
      }
    }
  />
</View>
`;

exports[`PHOTOINFO Renders photo info component 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "flex-start",
      "paddingBottom": 5,
      "paddingLeft": 3,
      "paddingTop": 5,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://s3.amazonaws.com/uifaces/faces/twitter/kfriedson/128.jpg",
      }
    }
    style={
      Object {
        "borderRadius": 15,
        "height": 30,
        "margin": 7,
        "width": 30,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignContent": "flex-start",
        "alignSelf": "center",
        "flexGrow": 1,
        "paddingLeft": 7,
      }
    }
  >
    josh.duplessis
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../src/imgs/more.png",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "height": 22,
        "marginLeft": 10,
        "marginRight": 10,
        "width": 22,
      }
    }
  />
</View>
`;
